author_generator.rb

```diff
@@ -71,9 +71,15 @@ def write_author_index(author_dir, author)
     def write_author_indexes
       author_dirs_written = []
 
+      if Jekyll::VERSION < '3.0.0'
+        $post_collection = self.posts
```

Comment:
@leenasn Fairly certain I added all of your requests, though I'm not sure what you meant by "also use the naming appropriately". If I missed something, let me know!